db/db_impl.h

```diff
@@ -696,6 +696,9 @@ class DBImpl : public DB {
   // The writer must be the leader in write_thread_ and holding mutex_
   Status WriteRecoverableState();
 
+  // Actual implementation of Close()
+  virtual Status CloseImpl();
```

Comment:
Should be protected

Reply:
What's the purpose of having a virtual DBImpl::CloseImpl()? Do we expect someone to override DBImpl? DB is an interface and DBImpl is the implementation. I don't see why someone wants to override DBImpl (may be except in unit tests) and even if they do why do they need to change CloseImpl()?

Reply:
Yes, its in the protected block.

Reply:
@siying I do see some sub-classes of DBImpl, but I agree with your point of not needing to override CloseImpl(). I'll change it to non-virtual.